src/plugins/core_plugin.ts

```diff
@@ -51,7 +51,14 @@ export class CorePlugin<State = any>
     uuidGenerator,
   }: CorePluginConfig) {
     super(stateObserver, dispatch, canDispatch);
-    range.addRangeProvider(this.adaptRanges.bind(this));
+
+    // Bind adaptRanges to a version of `this` where `dispatch` always throws
+    const thisWithThrowingDispatch = Object.create(this);
+    thisWithThrowingDispatch.dispatch = () => {
+      throw new Error("dispatch is not allowed in adaptRanges context");
+    };
+    range.addRangeProvider(this.adaptRanges.bind(thisWithThrowingDispatch));
```

Comment:
Add a test for this 🙂 

And maybe it should be the responsibility of the range plugin, rather than every other plugin ? With a boolean `isAdaptingRange` or something that would throw on a command `handle`  ... not sure tho.